### PR TITLE
New package: ktsu.BuildMonitor version 1.0.0.0

### LINKED ISSUE #160180: [Package Request]: ktsu.BuildMonitor
### How can we help?

None

### Please read and ensure the following

- [X] The installer meets the above requirements

### Please provide the following information

Download Page Url: https://github.com/ktsu-dev/BuildMonitor/releases/tag/1.0.0-alpha.1-win-x64
Publisher: ktsu
Package Name: BuildMonitor
Description: Tool for monitoring the status and progress of CI/CD builds.
Package Version: 1.0.0.0
Installer URL: https://github.com/ktsu-dev/BuildMonitor/releases/download/1.0.0-alpha.1-win-x64/BuildMonitor-1.0.0-alpha.1-win-x64.exe

### PR DESCRIPTION
**Pull request by matt-edmondson**
Checklist for Pull Requests
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Is there a linked Issue?

Manifests
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.6 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.6.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/160177)

Resolve #160180

### CODE REVIEW

**Comment by matt-edmondson** [comment]
https://github.com/microsoft/winget-pkgs/issues/160180

**Comment by stephengillie** [comment]
Automatic Validation ended with:

> Executable C:\Program Files\BuildMonitor\ktsu.BuildMonitor.exe returned exit code: -532462766

Automated error analysis suggests -532462766 may mean EXCEPTION_COMPLUS (Unhandled Exception)

(Automated response - build 886.)

**Comment by matt-edmondson** [comment]
@stephengillie any idea how I could find out what this exception is? When installing via the local manifest I am able to successfully install and run the application with no exceptions.

File: manifests/k/ktsu/BuildMonitor/1.0.0.0/ktsu.BuildMonitor.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Created using wingetcreate 1.6.1.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.6.0.schema.json
+
+PackageIdentifier: ktsu.BuildMonitor
+PackageVersion: 1.0.0.0
+InstallerType: nullsoft
+Installers:
+- Architecture: x64
+  InstallerUrl: https://github.com/ktsu-dev/BuildMonitor/releases/download/1.0.0-alpha.1-win-x64/BuildMonitor-1.0.0-alpha.1-win-x64.exe
+  InstallerSha256: 229EB9C552DFFCC2E395695991484C2E6ECC1CA660A02D940636934EDDCDEC82
```
Comment (Dragon1573) [inline manifests/k/ktsu/BuildMonitor/1.0.0.0/ktsu.BuildMonitor.installer.yaml:10]:
```suggestion
- Architecture: x64
  InstallerUrl: https://github.com/ktsu-dev/BuildMonitor/releases/download/1.0.0-alpha.1-win-x64/BuildMonitor-1.0.0-alpha.1-win-x64.exe
  InstallerSha256: 229EB9C552DFFCC2E395695991484C2E6ECC1CA660A02D940636934EDDCDEC82
  Dependencies:
    PackageDependencies:
    - PackageIdentifier: Microsoft.VCRedist.2015+.x64
```

_Microsoft Visual C++ Redistributable 2015-2022 (x64)_ is necessary here.

**Review by Dragon1573 — CHANGES_REQUESTED** [review]

**Comment by Dragon1573** [comment]
> When installing via the local manifest I am able to successfully install and run the application with no exceptions.

It would be better testing local manifests with `Tools/SandboxTest.ps1` in _Windows Sandbox_. Each developers may have their own development environment, and may have different packages pre-installed. Those pre-installed packages may affect the installation process of you local manifests.

_Windows Sandbox_ is a pure, brand new VM instance each time you launch it. The start point will always be the same on all developers. `Tools/SandboxTest.ps1` performs same actions on every _Sandbox_ instance, so you might be able to find some important key.

**Comment by stephengillie** [comment]
> @stephengillie any idea how I could find out what this exception is? When installing via the local manifest I am able to successfully install and run the application with no exceptions.

This same happens with a pair of other packages in the repo (`Ryujinx.Ryujinx` and `Ryujinx.Ryujinx.Ava`) on a regular basis. I've been manually swapping labels to allow them to pass, and would be happy to do the same here.

File: manifests/k/ktsu/BuildMonitor/1.0.0.0/ktsu.BuildMonitor.installer.yaml (hunk 1 above)
Comment (matt-edmondson) [reply manifests/k/ktsu/BuildMonitor/1.0.0.0/ktsu.BuildMonitor.installer.yaml:10]: I will put a step to install the redist in my installer, thankyou!

**Comment by matt-edmondson** [comment]
> duplicate

I spent today trying to get this working, but it seems as this is an OpenGL application using the Silk.NET library (same as Ryujinx) I had no luck getting it to actually run in the sandbox, perhaps why its failing in the automation.

I am adding the vcredist installation to my nsis installer and I'll submit another release.

Thankyou both.

**Comment by matt-edmondson** [comment]
I have remade the release including the vc redist package in the installer and updated the hash in the manifest.

As before, this installs and runs for me locally, as I have not been able to get opengl via Silk.NET to run in the sandbox yet.

**Comment by stephengillie** [comment]
- Possibly related: https://github.com/microsoft/winget-pkgs/issues/12906

File: manifests/k/ktsu/BuildMonitor/1.0.0.0/ktsu.BuildMonitor.locale.en-US.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.6.0.schema.json
+
+PackageIdentifier: ktsu.BuildMonitor
+PackageVersion: 1.0.0.0
+PackageLocale: en-US
+Publisher: ktsu
+PackageName: BuildMonitor
+License: MIT License
+Copyright: ktsu © 2024
+ShortDescription: ktsu.io.BuildMonitor
```
Comment (Trenly) [inline manifests/k/ktsu/BuildMonitor/1.0.0.0/ktsu.BuildMonitor.locale.en-US.yaml:11]: Please provide a better description of what this application is, what it does, or what it is used for

**Review by Trenly — CHANGES_REQUESTED** [review]

**Comment by matt-edmondson** [comment]
> * Possibly related: [Support OpenGL for package validation. #12906](https://github.com/microsoft/winget-pkgs/issues/12906)

Yeah, it seems like the same issue I was facing with validating it locally in my own sandbox, which makes sense.

File: manifests/k/ktsu/BuildMonitor/1.0.0.0/ktsu.BuildMonitor.locale.en-US.yaml (hunk 2 above)
Comment (matt-edmondson) [reply manifests/k/ktsu/BuildMonitor/1.0.0.0/ktsu.BuildMonitor.locale.en-US.yaml]: Addressed in: https://github.com/microsoft/winget-pkgs/pull/160177/commits/df3cf40664555290d5783b9d2c7702ecc42bf3c2

**Comment by stephengillie** [comment]
Manual Validation ended with:
> Application: ktsu.BuildMonitor.exe CoreCLR Version: 8.0.624.26715 .NET Version: 8.0.6 Description: The process was terminated due to an unhandled exception. Exception Info: Silk.NET.GLFW.GlfwException: ApiUnavailable: WGL: The driver does not appear to support OpenGL    at Silk.NET.GLFW.Glfw.<>c.<.cctor>b__143_0(ErrorCode errorCode, String description)    at Silk.NET.GLFW.Glfw.CreateWindow(Int32 width, Int32 height, String title, Monitor* monitor, WindowHandle* share)    at Silk.NET.Windowing.Glfw.GlfwWindow.CoreInitialize(WindowOptions opts)    at Silk.NET.Windowing.Internals.WindowImplementationBase.CoreInitialize(ViewOptions opts)    at Silk.NET.Windowing.Internals.ViewImplementationBase.Initialize()    at Silk.NET.Windowing.WindowExtensions.Run(IView view)    at ktsu.io.ImGuiApp.ImGuiApp.Start(String windowTitle, ImGuiAppWindowState initialWindowState, Action onStart, Action`1 onTick, Action onMenu, Action onWindowResized)    at ktsu.BuildMonitor.BuildMonitor.Main() in D:\a\1\s\BuildMonitor\BuildMonitor.cs:line 34  
> Faulting application name: ktsu.BuildMonitor.exe, version: 1.0.0.0, time stamp: 0x664796e4 
> Faulting module name: KERNELBASE.dll, version: 10.0.19041.1202, time stamp: 0xc9db1934 Exception code: 0xe0434352 Fault offset: 0x0000000000034f99 
> Faulting process id: 0x273c 
> Faulting application start time: 0x01dacbcd624488ae 
> Faulting application path: C:\Program Files\BuildMonitor\ktsu.BuildMonitor.exe 
> Faulting module path: C:\Windows\System32\KERNELBASE.dll Report Id: afc39ea1-facc-4841-b29e-0b04e724f848 
> Faulting package full name:  
> Faulting package-relative application ID:   

(Automated response - build 886.)

**Comment by matt-edmondson** [comment]
@stephengillie I'm not really sure how I should proceed with this as it seems like there isn't good enough support in windows sandbox to perform the validation. Is this the same situation all OpenGL applications are in? Do you know of any that pass validation I could compare?

**Comment by stephengillie** [comment]
Hi @matt-edmondson,

I was moving too fast and didn't stop to read the error.

- https://github.com/microsoft/winget-pkgs/issues/12906
- https://github.com/microsoft/winget-pkgs/issues/107639

**Review by stephengillie — APPROVED** [review]

**Comment by stephengillie** [comment]
Verify:
image

**Pull request merged**